src/host/gpunumber.jl

```diff
@@ -0,0 +1,42 @@
+# Custom GPU-compatible `Number` interface.
+struct GPUNumber{T <: AbstractGPUArray} <: AN.AbstractNumber{T}
+    val::T
+
+    function GPUNumber(val::T) where T <: AbstractGPUArray
```

Comment:
@pxl-th it might be a good idea to do a `dropdims` or `vec` in the constructor so this guy truly behaves like a number. The case I am thinking of is the following.

```julia
a = CUDA.rand(1,1,1,1) |> GPUNumber
x = CUDA.rand(4,4)
size(x .+ a) # want (4,4)
```

I haven't run your code but it seems like it would do something similar to
```julia
julia> CUDA.rand(1,1,1,1) .+ CUDA.rand(4,4) |> size
(4, 4, 1, 1)
```

Reply:
Indeed, thanks!